### DISCUSSION TITLE
Add mod shield to comments

### DISCUSSION THREAD

**Discussion by michael-tharrington** [forem/forem#17666] [Icebox]
I think it'd be ideal if we used the shield icon to link to mod actions on comments. 

We currently use the shield icon for mods (see fig 1): 
1. to expand the mod panel on articles 
2. to access the mod center (dev.to/mod) 

... and I think it'd be more consistent with our style if we expanded that functionality to comments too.

Currently, comments have a three-dot menu that is used to access "moderate", i.e. mod actions (see fig 2). When a user clicks on "moderate" for comments, they are taken to a separate page with various mod actions they can take on the comment (see fig 3).

I think it'd be ideal if when a moderator clicked on the shield icon for a comment, the mod panel would expand similar to how it does for articles. I believe that we could just change some of the language and available actions to fit comments instead of posts. (see rough sketch in fig 4)

**Screenshots**

Fig 1

![screenshot of shield icon and how it's used for articles and accessing the mod center](https://user-images.githubusercontent.com/16007075/168152972-6b74540e-15c1-41d2-b8fd-3a32310b6e5c.png)

Fig 2

![three-dot menu on comments that is used to access moderator actions](https://user-images.githubusercontent.com/16007075/168153574-ec3d701a-8426-4ae8-a730-5fb091c59a7d.png)

Fig 3

![separate page for moderating comments](https://user-images.githubusercontent.com/16007075/168153601-7d45783b-13ef-4abd-b4b4-ca62c7c30aaf.png)

Fig 4

![how the shield icon might work for moderating comments](https://user-images.githubusercontent.com/16007075/168154915-f505637d-062e-4515-a40f-80ef2b286255.png)

**Additional Context**

I see this request as being very similar to https://github.com/forem/forem/discussions/17311 ... it's all about design consistency and should help bring clarity to mods as they'll know that clicking the shield takes them to mod-level actions.

**Comment by amywtlin** [comment]
@ludwiczakpawel How do you feel about the approach of moderating comments, design-wise?

**Reply by ludwiczakpawel** [reply]
I think it's fine for now - lets just adjust what we have so it reflects the suggestions from @michael-tharrington. 

We've been thinking about making that Mod Panel a bit nicer, cleaner, more intuitive and easier to use but it's not a top of the priority right now. So we will have to address visual side of things at some point anyway. Just not now. And while we do that, we could also add a bit of love to the Comment variant of that panel. But yea, for now - "anything" is good.

**Reply by amywtlin** [reply]
I agree. I don't think this one is of high priority, but I was curious about how you feel about the design direction. @michael-tharrington I am going to put this one in icebox for now, but I am glad we keep what you have in mind documented so we can come back to it easily when we are ready.

**Reply by michael-tharrington** [reply]
Good stuff! That makes sense. No rush as far as I'm concerned. 🙂
